modules/agar/src/main/ts/ephox/agar/api/Mouse.ts

```diff
@@ -46,17 +55,24 @@ const cMouseMove = cMouseMoveWith({ });
  * @deprecated use cMouseOutWith({ }) instead */
 const cMouseOut = cMouseOutWith({ });
 
+const triggerOn = <T extends Element>(container: SugarElement<Node>, selector: string, action: (ele: SugarElement<T>) => void): SugarElement<T> => {
+  const ele = UiFinder.findIn(container, selector).getOrDie();
+  action(ele);
+  return ele;
+};
+
 // Work with selectors
 const sTriggerOn = <T, U extends Element>(container: SugarElement<Node>, selector: string, action: (ele: SugarElement<U>) => void) =>
-  Chain.asStep<T, SugarElement<Node>>(container, [ Chain.async((container, next, die) => {
-    UiFinder.findIn(container, selector).fold(
-      () => die('Could not find element: ' + selector),
-      (ele) => {
-        action(ele);
-        next(container);
-      }
-    );
-  }) ]);
+  Step.sync<T>(() => triggerOn(container, selector, action));
+
+const clickOn = <T extends HTMLElement>(container: SugarElement<Node>, selector: string): SugarElement<T> =>
+  triggerOn<T>(container, selector, Clicks.trigger);
+
+const hoverOn = <T extends Element>(container: SugarElement<Node>, selector: string): SugarElement<T> =>
+  triggerOn<T>(container, selector, Clicks.mouseOver({ }));
```

Comment:
Should we use the const declared earlier in here?

```suggestion
  triggerOn<T>(container, selector, mouseOver);
```